lib/common/variadic/uint64OrHash_test.go

```diff
@@ -72,22 +53,22 @@ func TestNewUint64OrHashFromBytes(t *testing.T) {
 			description:     "block request with Block Number int type 1",
 			targetHash:      buf,
 			targetFirstByte: 1,
-			expectedType:    (uint64)(0),
+			expectedType:    (uint32)(0),
 		},
 	} {
 		t.Run(x.description, func(t *testing.T) {
 			data := append([]byte{x.targetFirstByte}, x.targetHash...)
 
-			uint64OrHash := NewUint64OrHashFromBytes(data)
+			val := NewUint32OrHashFromBytes(data)
 			require.NoError(t, err)
-			require.NotNil(t, uint64OrHash)
-			require.IsType(t, x.expectedType, uint64OrHash.Value())
-			if x.expectedType == (uint64)(0) {
-				startingBlockByteArray := make([]byte, 8)
-				binary.LittleEndian.PutUint64(startingBlockByteArray, uint64OrHash.Value().(uint64))
+			require.IsType(t, x.expectedType, val.Value())
+
+			if x.expectedType == (uint32)(0) {
```

Comment:
nit that should work fine

```suggestion
			if x.expectedType == 0 {
```

Reply:
sadly doesn't work :( it think it's not equal

Reply:
Ah my bad, I didn't notice `x.expectedType` was an `interface{}`, sorry about that.